pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -278,6 +278,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(
 				bindToPort:     false,
 			}
 
+			if endpoint.ServicePort.Protocol == protocol.Unsupported {
+				endpoint.ServicePort.Protocol = protocol.TCP
```

Comment:
Can you explain why we need these to be set to TCP?

Reply:
We assume unknown ports to be TCP in current code path. e.g., for port "test", the conversion output is protocol.TCP. After refactoring, the conversion output is protocol.Unsupported. So we need to set Unsupported to TCP somewhere in order not to break current code path.

Reply:
Which code paths does it break if we remove this altogether? And then a follow-up quesiton ... *should* they be broken by this or is the code perhaps too fragile?

Reply:
For Unsupported protocol, pilot will not build listener on that port. If users use port "test", before they expect to have a TCP proxy on that port but currently they get nothing.

Reply:
Discussed offline.